src/vm/jitinterfacegen.cpp

```diff
@@ -246,9 +248,11 @@ void InitJITHelpers1()
 
                 ECall::DynamicallyAssignFCallImpl(GetEEFuncEntryPoint(AllocateStringFastMP), ECall::FastAllocateString);
             }
+#endif // FEATURE_IMPLICIT_TLS
         }
         else
         {
+#ifndef FEATURE_IMPLICIT_TLS
```

Comment:
The UP allocation helpers do not access TLS, so it is not logical to put them under FEATURE_IMPLICIT_TLS. I am wondering whether it would be better to structure it this way: 

```
#ifdef FEATURE_PAL
    _ASSERTE(GCHeap::UseAllocationContexts());
    SetJitHelperFunction(CORINFO_HELP_NEWARR_1_VC, JIT_NewArr1VC_MP_FastPortable); 
#else
    if (GCHeap::UseAllocationContexts()) 
    {
        .. the existing code...
    }
#endif // FEATURE_PAL
```